golem-common/src/model/mod.rs

```diff
@@ -777,12 +793,416 @@ pub fn parse_function_name(name: &str) -> ParsedFunctionName {
     }
 }
 
+// #[derive(Clone, Debug, Serialize, Deserialize, Encode, Decode)]
+// pub struct WorkerNameFilter {
+//     pub comparator: StringFilterComparator,
+//     pub value: String,
+// }
+//
+// #[derive(Clone, Debug, Serialize, Deserialize, Encode, Decode)]
+// pub struct WorkerStatusFilter {
+//     pub value: WorkerStatus
+// }
+//
+//
+// #[derive(Clone, Debug, Serialize, Deserialize, Encode, Decode)]
+// pub struct WorkerVersionFilter {
+//     pub comparator: FilterComparator,
+//     pub value: i32,
+// }
+//
+// #[derive(Clone, Debug, Serialize, Deserialize, Encode, Decode)]
+// pub struct WorkerVersionEnvFilter {
+//     pub name: String,
+//     pub comparator: StringFilterComparator,
+//     pub value: String
+// }
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, Encode, Decode)]
+pub enum WorkerFilter {
+    Name {
+        comparator: StringFilterComparator,
+        value: String,
+    },
+    Status {
+        value: WorkerStatus,
+    },
+    Version {
+        comparator: FilterComparator,
+        value: i32,
+    },
+    // CreatedAt {
```

Comment:
Once https://github.com/golemcloud/golem/issues/238 is done we will have creation data (because each oplog entry has a timestamp) so I think we can "prepare" for it already